core/runtime/src/main/java/io/quarkiverse/flow/recorders/WorkflowApplicationRecorder.java

```diff
@@ -124,4 +127,14 @@ private void injectConfigManager(final ArcContainer container, final WorkflowApp
         }
     }
 
+    private void injectHttpClientProvider(final ArcContainer container, final WorkflowApplication.Builder builder) {
+        final HttpClientProvider httpClientProvider = container.instance(HttpClientProvider.class).get();
+        LOG.info("Flow: Bound HttpClientProvider bean: {}", httpClientProvider.getClass().getName());
+        builder.withAdditionalObject(HttpClientResolver.HTTP_CLIENT_PROVIDER, ((workflowContextData, taskContextData) -> {
+            final String workflowName = workflowContextData.definition().workflow().getDocument().getName();
+            final String taskName = taskContextData.taskName();
+            return httpClientProvider.clientFor(workflowName, taskName);
```

Comment:
What I was trying to say in my prev comment is that you can return null here in case there is not a client defined fro that workflow name and taskname. 
Also, more importantly, the name does not univocally identify a Workflow an/or a Task, you should use [WorkflowDefinition ](https://github.com/serverlessworkflow/sdk-java/blob/main/impl/core/src/main/java/io/serverlessworkflow/impl/WorkflowDefinitionId.java) class and the `TaskContext.position()`
Yes, that requires changes in the way the properties are setup, but better late than never

Reply:
@fjtirado I spent some time thinking about this. The tradeoff of having a naming clash because I have in the same app one workflow with the same name, different namespaces, different versions is too low to sacrifice user experience.

It's way easier to express `quarkus.flow.http.client.myworkflow.mytask` rather than `quarkus.flow.http.client.mynamespace-mywoprkflow-myversion.mytask`.